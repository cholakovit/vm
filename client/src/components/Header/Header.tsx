// Styled Elements
import {
  MaterialUISwitch,
  VmFormControlLabel,
  VmAppBar,
  HeaderContainer
} from './Header.styles';

// Hooks
import React, { useContext } from 'react';

// Context for the Theme
import { ColorModeContext } from '../../helper/Context';

// Types
import type { colorModeProps } from '../../types';

const Header = () => {
  const colorMode: colorModeProps = useContext(ColorModeContext) || {};

  return (
    <VmAppBar>
      <HeaderContainer>
        <VmFormControlLabel
          label=""
          onClick={colorMode.toggleColorMode}
          control={<MaterialUISwitch defaultChecked />}
          data-testid="button"
        />
      </HeaderContainer>
    </VmAppBar>
  );
};

export default Header;
