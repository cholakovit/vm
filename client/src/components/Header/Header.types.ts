// for the Theme
export interface colorModeProps {
  toggleColorMode?: any;
};