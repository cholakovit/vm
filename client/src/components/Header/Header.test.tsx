// Header.test.tsx
import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

jest.mock('../../helper/Context', () => ({
  ColorModeContext: {
    Consumer: ({ children }: any) => children({}),
  },
}));

describe('Header component', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
