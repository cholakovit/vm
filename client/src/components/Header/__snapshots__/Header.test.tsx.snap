// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders without crashing 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1sf53z6-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1rcwb2f-MuiContainer-root"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-u81gj2-MuiFormControlLabel-root"
      data-testid="button"
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeMedium css-yhmnjz-MuiSwitch-root"
      >
        <span
          className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            defaultChecked={true}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      />
    </label>
  </div>
</header>
`;
